src/perl/lib/wtsi_clarity/epp/generic/stamper.pm

```diff
@@ -173,25 +180,29 @@ sub _build__analytes {
     ##use critic
     my $analyte_dom   = $self->fetch_and_parse($url);
     my $container_url = $analyte_dom->findvalue($CONTAINER_PATH);
+
     if (!$container_url) {
       croak qq[Container not defined for $url];
     }
 
+    my @control_flag = $analyte_dom->findnodes($CONTROL_PATH);
+
+    # By default we do not want to stamp controls (unless it's a shadow plate)
+    if (@control_flag && !$self->with_controls && !$self->shadow_plate) {
```

Comment:
Sorry, I wrote the wrong thing, I meant:
if with_controls => 0 and shadow_plate => 1 it will add the control, because the expression's result is false.